packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/session_manager.dart

```diff
@@ -0,0 +1,203 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/sdk/pinpoint.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:intl/intl.dart';
+import 'package:meta/meta.dart';
+
+/// Enum indicating the state of the app session
+enum SessionState {
+  /// App is not active (backgrounded)
+  inactive,
+
+  /// App is running (foregrounded)
+  active,
+
+  /// App is paused
+  paused,
+}
+
+/// Function that listens to Session lifecycle updates
+typedef OnSessionUpdated = void Function(Session);
+
+/// {@template amplify_analytics_pinpoint_dart.session_manager}
+/// Manage creation and deletion of current Session.
+///
+/// Updates session based on App foreground/background events
+/// Used for provisioning Events sent to Pinpoint with a Session.
+/// {@endtemplate}
+class SessionManager {
+  /// {@template amplify_analytics_pinpoint_dart.session_manager_constructor}
+  /// {@macro amplify_analytics_pinpoint_dart.session_manager}
+  ///
+  /// The parameters: [onSessionStart] and [onSessionEnd] are callbacks to
+  /// allow parent classes to react to session events.
+  /// {@endtemplate}
+  @visibleForTesting
+  SessionManager({
+    required String fixedEndpointId,
+    AppLifecycleProvider? appLifecycleProvider,
+    required OnSessionUpdated onSessionStart,
+    required OnSessionUpdated onSessionEnd,
+  })  : _fixedEndpointId = fixedEndpointId,
+        _appLifecycleProvider = appLifecycleProvider,
+        _onSessionStart = onSessionStart,
+        _onSessionEnd = onSessionEnd {
+    // Start session by _executeStart()
+    // By the time we start observing on lifecycleObserver,
+    // we have missed the first onForeground event
```

Comment:
What is the implication of this? It's not clear why this is mentioned here.

Reply:
Removed and added comment to the actual method